include/picongpu/plugins/binning/Binner.hpp

```diff
@@ -431,6 +468,60 @@ namespace picongpu
                         mapper);
             }
 
+            template<typename T_Species>
+            struct BinLeavingParticles
+            {
+                using Species = pmacc::particles::meta::
+                    FindByNameOrType_t<VectorAllSpecies, T_Species, pmacc::errorHandlerPolicies::ReturnType<void>>;
+
+                auto operator()(int32_t direction) const -> void
+                {
+                    if constexpr(!std::is_same_v<void, Species>)
+                    {
+                        auto& dc = Environment<>::get().DataConnector();
+                        auto particles = dc.get<Species>(Species::FrameType::getName());
+                        auto particlesBox = particles->getDeviceParticlesBox();
+                        auto binningBox = histBuffer->getDeviceBuffer().getDataBox();
+
+                        auto mapperFactory = particles::boundary::getMapperFactory(*particles, direction);
```

Comment:
please use the exchange mapper, e.g.

```
ExchangeMapping<GUARD, MappingDesc> mapper(mappingDesc, exchangeType);
```

to avoid that you take particle moving to `TOP+RIGHT` into the notification for particles moving `RIGHT`

Reply:
Examples of how to use can be found in `pmacc/field/operatons/AddExchangeToBorder`

Reply:
using ExchangeMapper is not possible because we delete particles at the particle boundary which is not nessesary the guard.